src/components/icon-button/IconButton.styles.ts

```diff
@@ -0,0 +1,329 @@
+import styled, { css } from 'styled-components';
+import { FlattenSimpleInterpolation } from 'styled-components/ts3.6';
+
+import { TIconButtonSize, IconButtonLayer, IconButtonState } from './IconButton.types';
+import {
+    ICON_BUTTON_SIZES,
+    ICON_BUTTON_DIMENSIONS,
+    ICON_BUTTON_PADDING,
+    ICON_BUTTON_SPACING,
+} from './IconButton.constants';
+
+function generateSizingStyles(size: TIconButtonSize): FlattenSimpleInterpolation {
+    return css`
+        &[data-size='${size}'] {
+            --horizontal-padding: ${ICON_BUTTON_PADDING[size]}px;
+            --size: ${ICON_BUTTON_DIMENSIONS[size]}px;
+            --child-spacing: ${ICON_BUTTON_SPACING[size]}px;
+        }
+    `;
+}
+
+function getRBGColour({ colour, opacity }: IconButtonLayer): string {
+    if (opacity && opacity >= 0 && opacity < 1) {
+        return `rgba(var(${colour}), ${opacity})`;
+    }
+
+    return `rgb(var(${colour}))`;
+}
+
+function getLayerColours(layer: IconButtonLayer): string {
+    return `--colour-${layer.name}: ${getRBGColour(layer)};`;
+}
+
+function generateStateStyles(stateStyles: IconButtonState[]): FlattenSimpleInterpolation {
+    return css`
+        ${stateStyles
+            .map(({ selector, layers }: IconButtonState) => {
+                if (selector === 'base') {
+                    return `
+                        ${layers.map(getLayerColours).join('\n')}
+                    `;
+                }
+
+                return `
+                    &:${selector} {
+                        ${layers.map(getLayerColours).join('\n')}
+                    }
+                `;
+            })
+            .join('\n')};
+    `;
+}
+
+const StyledIconButton = styled.button`
+    // set component colours
+    --colour-primary-rgb: var(--button-colour-primary-rgb, 61, 60, 64);
+    --colour-inverted-primary-rgb: var(--button-colour-inverted-primary-rgb, 255, 255, 255);
+    --colour-secondary-rgb: var(--button-colour-secondary-rgb, 22, 109, 224);
+    --colour-inverted-secondary-rgb: var(--button-colour-inverted-secondary-rgb, 20, 93, 191);
+    --colour-focus-rgb: var(--button-colour-focus-rgb, 22, 109, 224);
```

Comment:
do we use the british `colour` or the american `color`?

IMO we should use `color` since it is used in CSS the same way.

Reply:
Since I live right next to the US, I like to be a rebel :P

That said, I'm find with 'color'.

Reply:
I actually had the same thoughts I prefer colour but I know most people are used with the CSS prop version

Reply:
i can add a eslint regex to help identify such things, if it helps! :D